### PR TITLE
[cloudflare-ingress-controller] Add helmfile

### PR DESCRIPTION
**Pull request by osterman**
## what
* Add Cloudflare ingress controller (aka Argo / acsess)
* Added Ingress configs for Grafana
* Added Ingress configs for Prometheus
* Added Ingress configs for Alerts Manager
* Added Ingress configs for K8s Dashboard
* Added Traefik ingress controller
* Added workaround for ingress and AWS elastic search Kibana

## why
* Expose services inside Kubernetes securely and speedily using Argo tunnels
* Publish Grafana with Cloudflare Argo auth
* Publish Prometheus with Cloudflare Argo auth
* Publish Alers Manager with Cloudflare Argo auth
* Publish k8s Dashboard with Cloudflare Argo auth
* AWS elastic cache Kibana allow asses by https, Argo ingress can use HTTPS-only as upstream, Traefik is "reverse https terminator"
* Publish Kibana with cloudflare Argo auth

### CODE REVIEW

File: helmfile.d/0350.cloudflare-ingress-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- name: "argo"
+  namespace: "kube-system"
+  labels:
+    chart: "cloudflare-ingress-controller"
+    component: "ingress"
+    namespace: "kube-system"
+    vendor: "cloudflare"
+    default: "false"
+  chart: "cloudposse/cloudflare-ingress-controller"
+  version: "0.5.1"
```
Comment (osterman) [inline helmfile.d/0350.cloudflare-ingress-controller.yaml]: upgrade to 0.5.3

File: helmfile.d/0350.cloudflare-ingress-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+helmDefaults:
+  args:
```
Comment (osterman) [inline helmfile.d/0350.cloudflare-ingress-controller.yaml]: Update to use helmfile native args.

File: helmfile.d/0350.cloudflare-ingress-controller.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    - "--wait"
+    - "--timeout=600"
+    - "--force"
+    - "--reset-values"
+
+repositories:
+# Cloud Posse incubator repo of helm charts
+- name: "cloudposse-incubator"
+  url: "https://charts.cloudposse.com/incubator/"
```
Comment (osterman) [inline helmfile.d/0350.cloudflare-ingress-controller.yaml]: Use 
```
helm repo add cloudflare https://cloudflare.github.io/helm-charts
helm repo update
```

File: helmfile.d/0350.cloudflare-ingress-controller.yaml (hunk 1 above)
Comment (osterman) [reply helmfile.d/0350.cloudflare-ingress-controller.yaml]: Upgrade to latest

File: helmfile.d/0350.cloudflare-ingress-controller.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    default: "false"
+  chart: "cloudposse/cloudflare-ingress-controller"
+  version: "0.5.1"
+  values:
+    # Default values for cloudflare-argo-ingress.
+    # replicaCount is used only when cloudflare loadbalancing is enabled
+    - replicaCount: 1
+
+      image:
+        repository: "gcr.io/stackpoint-public/argot"
```
Comment (osterman) [inline helmfile.d/0350.cloudflare-ingress-controller.yaml]: Probably wrong.

File: helmfile.d/0350.cloudflare-ingress-controller.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      resources:
+        limits:
+          cpu: "100m"
+          memory: "128Mi"
+        requests:
+          cpu: "100m"
+          memory: "128Mi"
+
+      rbac:
+        install: false
```
Comment (osterman) [inline helmfile.d/0350.cloudflare-ingress-controller.yaml]: Parameterize

File: releases/cloudflare-ingress-controller.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          repository: "gcr.io/cloudflare-registry/argo-tunnel"
+          tag: "0.6.2"
+          pullPolicy: "IfNotPresent"
+        ingressClass: argo-tunnel
+        logLevel: 4
+        replicaCount: 1
+        defaultOriginSecret: 'kube-system/argo-secret-monochart-files-default'
+        resources:
+          limits:
+            cpu: 100m
```
Comment (aknysh) [inline releases/cloudflare-ingress-controller.yaml]: parameterize? (as we started doing in other helmfiles)

File: releases/kibana-elastic-search.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  wait: true
+  values:
+  - resources:
+    - apiVersion: extensions/v1beta1
+      kind: Ingress
+      metadata:
+        namespace: kube-system
+        name: kibana-aws
+        annotations:
+          kubernetes.io/ingress.class: {{ env "KIBANA_INTERNAL_CLASS" | default "argo-tunnel" }}
```
Comment (aknysh) [inline releases/kibana-elastic-search.yaml:37]:
```suggestion
          kubernetes.io/ingress.class: {{ env "KIBANA_INTERNAL_INGRESS_CLASS" | default "argo-tunnel" }}
```

File: releases/traefik-ingress.yaml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  chart: "stable/traefik"
+  version: "1.59.1"
+  wait: true
+  values:
+  - image: traefik
+    imageTag: 1.7.7
+    serviceType: {{ env "TRAEFIK_INGESS_SERVICE_TYPE" | default "LoadBalancer" }}
+    externalTrafficPolicy: ''
+    replicas: {{ env "TRAEFIK_REPLICA_COUNT" | default "1" }}
+    cpuRequest: 100m
```
Comment (aknysh) [inline releases/traefik-ingress.yaml]: parameterize?

File: releases/traefik-ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+      storeAcme: false
+      importAcme: false
+    gzip:
+      enabled: true
+    traefikLogFormat: json
+    rbac:
+      enabled: {{ env "RBAC_ENABLED" | default "false" }}
+    deployment:
+      hostPort:
+        httpEnabled: false
```
Comment (aknysh) [inline releases/traefik-ingress.yaml]: everything is disabled? (might be an incorrect question, just looks strange :)

File: releases/dashboard.yaml [hunk 10]
```diff
@@ -33,6 +33,15 @@ releases:
         ### Optional: KUBERNETES_DASHBOARD_IMAGE_TAG;
         tag: '{{ env "KUBERNETES_DASHBOARD_IMAGE_TAG" | default "v1.10.0" }}'
         pullPolicy: "IfNotPresent"
+      enableInsecureLogin: '{{ env "KUBERNETES_DASHBOARD_ENABLE_INSECURE_LOGIN" | default "false"}}'
```
Comment (aknysh) [inline releases/dashboard.yaml]:
```suggestion
      enableInsecureLogin: {{ env "KUBERNETES_DASHBOARD_ENABLE_INSECURE_LOGIN" | default "false"}}
```

**Review by aknysh — COMMENTED** [review]
see comments

File: releases/traefik-ingress.yaml [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
...UHZPTk5MZnNBbXc3dUhUKzNiSzEzSUloeDI3ZmV2cXVWcENzNDFQNnBzdStWTG4yCjVIRHk0MXRoQkN3T0wrTithbGJ0ZktTcXM3TEFzM25RTjFsdHpITHZ5MGE1RGhkakpUd2tQclQrVXhwb0tCOUgKNFpZazErRUR0N09QbGh5bzM3NDFRaE4vSkNZK2RKbkFMQnNValFJREFRQUJvNEgzTUlIME1CMEdBMVVkRGdRVwpCQlJwZVc1dFhMdHh3TXJvQXM5d2RNbTUzVVVJTERDQnhBWURWUjBqQklHOE1JRzVnQlJwZVc1dFhMdHh3TXJvCkFzOXdkTW01M1VVSUxLR0JsYVNCa2pDQmp6RUxNQWtHQTFVRUJoTUNWVk14RVRBUEJnTlZCQWdUQ0VOdmJHOXkKWVdSdk1SQXdEZ1lEVlFRSEV3ZENiM1ZzWkdWeU1SUXdFZ1lEVlFRS0V3dEZlR0Z0Y0d4bFEyOXljREVMTUFrRwpBMVVFQ3hNQ1NWUXhGakFVQmdOVkJBTVVEU291WlhoaGJYQnNaUzVqYjIweElEQWVCZ2txaGtpRzl3MEJDUUVXCkVXRmtiV2x1UUdWNFlXMXdiR1V1WTI5dGdna0FrQVpDVXlZTFNUSXdEQVlEVlIwVEJBVXdBd0VCL3pBTkJna3EKaGtpRzl3MEJBUVVGQUFPQ0FRRUFjR1hNZms4TlpzQit0OUtCemwxRmw2eUlqRWtqSE8wUFZVbEVjU0QyQjRiNwpQeG5NT2pkbWdQcmF1SGI5dW5YRWFMN3p5QXFhRDZ0YlhXVTZSeENBbWdMYWpWSk5aSE93NDVOMGhyRGtXZ0I4CkV2WnRRNTZhbW13QzFxSWhBaUE2MzkwRDNDc2V4N2dMNm5KbzdrYnIxWVdVRzN6SXZveGR6OFlEclpOZVdLTEQKcFJ2V2VuMGxNYnBqSVJQNFhac25DNDVDOWdWWGRoM0xSZTErd3lRcTZoOVFQaWxveG1ENk5wRTlpbVRPbjJBNQovYkozVktJekFNdWRlVTZrcHlZbEpCemRHMXVhSFRqUU9Xb3NHaXdlQ0tWVVhGNlV0aXNWZGRyeFF0aDZFTnlXCnZJRnFhWng4NCtEbFNDYzkzeWZrL0dsQnQrU0tHNDZ6RUhNQjlocVBiQT09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
```
Comment (osterman) [inline releases/traefik-ingress.yaml]: What is this?

File: releases/traefik-ingress.yaml [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
...YVFQcXMrRHZ2REpzQ3gyZnRudk1WWVJKcVliaGt6YnpsVQowSHBCVnk3NmE3WmF6Umxhd3RGZ3ljMlpyQThpM0F3K3J6d1pQclNJeWNieC9nUVduRzZlbFF1Y0FFVWdXODRNCkdSbXhKUGlmOGRQNUxsZXdRalFjUFJwZVoxMzlYODJreGRSSEdma1pscHlXQnFLajBTWExRSEVDZ1lFQXcybkEKbUVXdWQzZFJvam5zbnFOYjBlYXdFUFQrbzBjZ2RyaENQOTZQK1pEekNhcURUblZKV21PeWVxRlk1eVdSSEZOLwpzbEhXU2lTRUFjRXRYZys5aGlMc0RXdHVPdzhUZzYyN2VrOEh1UUtMb2tWWEFUWG1NZG9xOWRyQW9INU5hV2lECmRSY3dEU2EvamhIN3RZV1hKZDA4VkpUNlJJdU8vMVZpbDBtbEk5MENnWUVBb2lsNkhnMFNUV0hWWDNJeG9raEwKSFgrK1ExbjRYcFJ5VEg0eldydWY0TjlhYUxxNTY0QThmZGNodnFiWGJHeEN6U3RxR1E2cW1peUU1TVpoNjlxRgoyd21zZEpxeE14RnEzV2xhL0lxSzM0cTZEaHk3cUNld1hKVGRKNDc0Z3kvY0twZkRmeXZTS1RGZDBFejNvQTZLCmhqUUY0L2lNYnpxUStQREFQR0YrVHFFQ2dZQmQ1YnZncjJMMURzV1FJU3M4MHh3MDBSZDdIbTRaQVAxdGJuNk8KK0IvUWVNRC92UXBaTWV4c1hZbU9lV2Noc3FCMnJ2eW1MOEs3WDY1NnRWdGFYay9nVzNsM3ZVNTdYSFF4Q3RNUwpJMVYvcGVSNHRiN24yd0ZncFFlTm1XNkQ4QXk4Z0xiaUZhRkdRSDg5QWhFa0dTd1d5cWJKc2NoTUZZOUJ5OEtUCkZaVWZsUUtCZ0V3VzJkVUpOZEJMeXNycDhOTE1VbGt1ZnJxbllpUTNTQUhoNFZzWkg1TXU0MW55Yi95NUUyMW4KMk55d3ltWGRlb3VJcFZjcUlVTXl0L3FKRmhIcFJNeVEyWktPR0QyWG5YaENNVlRlL0FQNDJod294Nm02QkZpQgpvemZFa2wwak5uZmREcjZrL1p2MlQ1TnFzaWxaRXJBQlZGOTBKazdtUFBIa0Q2R1ZMUUJ4Ci0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg==
```
Comment (osterman) [inline releases/traefik-ingress.yaml]: This looks bad

File: releases/traefik-ingress.yaml (hunk 9 above)
Comment (goruha) [reply releases/traefik-ingress.yaml]: This is default values.

File: releases/dashboard.yaml (hunk 10 above)
Comment (goruha) [reply releases/dashboard.yaml]: Done

**Review by osterman — COMMENTED** [review]
LGTM +2

**Review by goruha — APPROVED** [review]

**Pull request merged**